core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -524,18 +527,27 @@ object ConsumerGroupCommand extends Logging {
     val describeOpt = parser.accepts("describe", DescribeDoc)
     val deleteOpt = parser.accepts("delete", DeleteDoc)
     val newConsumerOpt = parser.accepts("new-consumer", NewConsumerDoc)
+    val groupInitializationTimeoutMsOpt = parser.accepts("group-init-timeout", GroupInitializationTimeoutMsDoc)
+                                                 .withRequiredArg
+                                                 .describedAs("group initialization timeout (ms)")
+                                                 .ofType(classOf[Long])
+                                                 .defaultsTo(1000)
     val commandConfigOpt = parser.accepts("command-config", CommandConfigDoc)
                                   .withRequiredArg
                                   .describedAs("command config property file")
                                   .ofType(classOf[String])
     val options = parser.parse(args : _*)
 
     val useOldConsumer = options.has(zkConnectOpt)
+    val describeOptPresent = options.has(describeOpt)
 
     val allConsumerGroupLevelOpts: Set[OptionSpec[_]] = Set(listOpt, describeOpt, deleteOpt)
 
     def checkArgs() {
       // check required args
+      if (options.has(groupInitializationTimeoutMsOpt) && (!describeOptPresent || useOldConsumer))
```

Comment:
It's kind of confusing to have a parameter which is only used for some options. Why don't we just ignore it if we don't need it? Perhaps we can log a debug-level warning if we must.